### ISSUE TITLE
switch license?

### ISSUE THREAD

**Issue by 517qf** [renerocksai/sublimeless_zk#35]
I'm not a lawyer and don't really know anything about licenses or programming. I just post as a precaution for the unlikely event that a) my points are correct and b) you accidently missed them. 

in your repo is a file LICENSE that says sublimeless_zk is licensed unter the MIT license.

pyqt5 is gplv3. you seem to use pyqt5. Your work seems to be a derived work. 

pyqt5 has a different license that pyqt4 which seems to offer an exception for different foss licenses. I can't find a similar excepton for pyqt5. GPL may not be relicensed under MIT.

So using MIT seems like a bad idea.

http://pyqt.sourceforge.net/Docs/PyQt5/introduction.html#license

https://softwareengineering.stackexchange.com/questions/235547/under-what-license-may-this-pyqt-based-hello-world-app-be-distributed

**Comment by renerocksai** [comment]
Thanks for pointing this out. I was more concerned about being permissive with my code, and because pyqt5 is now on pypi, I assumed it was permissive, too. The GPL is a viral license, so yes, I have to change the license. But not because my work is a derived work. It's a combined work. Anyway. Licenses.... Can be a pain. 

However, if the lawyers came threatening me for having MIT in the git history, I would probably be so pissed off that I'd just delete the repo. In which case all the GPL's supposed foresight might neither be beneficial to pyqt5, which is specifically built for programmers (and I used it exactly for what it was built for), nor sublimeless_zk, which is specifically built for users. It's totally pointless to publish a library that people can only use for writing programs for themselves and not share them because of potential free-software-license violations. That's so anti-free-software it hurts. Let's hope GPLing will be enough.

So, now I wonder, is Anaconda GPL? Can a single python package make a whole python ditribution GPL? If it can do so with my few source files, I suppose it can. What about the other packages then? I see potential for a lot of license conflicts. OMG, licenses, licenses.

**Comment by renerocksai** [comment]
>   A compilation of a covered work with other separate and independent
works, which are not by their nature extensions of the covered work,
and which are not combined with it such as to form a larger program,
in or on a volume of a storage or distribution medium, is called an
"aggregate" if the compilation and its resulting copyright are not
used to limit the access or legal rights of the compilation's users
beyond what the individual works permit.  Inclusion of a covered work
in an aggregate does not cause this License to apply to the other
parts of the aggregate.

Aha, so Anaconda seems to be safe 😄

**Comment by 517qf** [comment]
> Let's hope GPLing will be enough.

Why shouldn't it (but again I don't know anything)? I don't know many programs that use pyqt5 - the only example I know is the upcoming Anki 2.1 which is already widely used. The only thing I see is that Anki like other gpl software has an About/help menu that says licensed under GPL. So adding this might be a good idea. I think the GPLv3 is a good licence for your project. It might only limit you if you wanted to sell it (which in practice probably requires code obfuscation - otherwise why pay ...) or you wanted to publish this in some app stores. 

----

About the old licence: This is isolated in a single commit - can't you just remove [this commit](https://github.com/renerocksai/sublimeless_zk/commit/8c1a34135b3ecf93be0a43be1f1ce5602b01928b#diff-9879d6db96fd29134fc802214163b95a)? This seems like very little work: https://stackoverflow.com/questions/448919/how-can-i-remove-a-commit-on-github

----

Maybe the repository itself is not such a big problem because it is just code - isn't this similar to scenario 3](https://softwareengineering.stackexchange.com/a/235846)? A bigger problem might be the compiled versions you provide? No one neeeds older compiled versions, why not just remove them ... and at your pace a version 0.6 is hopefully coming soon ...

**Comment by 517qf** [comment]
offtopic: did you leave ST towards VSCode and do you already have plans for the future of sublime_ZK ? 

P.S.: I think both names only work for insiders - the intersection of zettelkasten aficionados and ST users is very small. There are probably many people who would be really happy with your programs but who will never know about them because of [satisficing](https://en.wikipedia.org/wiki/Satisficing) - they'll just search for "wiki", or "markdown notes", or "markdown wiki". I think your name prevents your software from showing high up in such a search and they will never know about the true power of a zettelkasten. So maybe adding wiki to the names might be an option?

**Comment by renerocksai** [comment]
Good points you are raising here. I can recall a time when Qt3 was GPLed, that provided such a menu automatically 😄 . Erasing the commit, I can try that. But erasing history is something I am not too comfortable with. No selling, no app store hassles, so it should be fine. Deleting the old releases is easy, they are superseded by newer releases anyway.

**Comment by renerocksai** [comment]
Regarding VSCode. That was just committed by accident 😄 . I recently switched from my paid version of PyCharm to VSCode. It is more lightweight, does everything I need, has a command palette 😄 and is just as cross-platform. I really like it. It's like a combination of the best of SublimeText and PyCharm. Now, that is for "real" python development.

Sublime_ZK, on the other hand, was entirely developed using SublimeText---which I use(d) mostly for my Markdown and Zettelkasten files. I will continue to maintain sublime_ZK, fix bugs and add requested features as I see fit, just as I did before. Currently I am more invested in sublimeless_ZK, though. I'll add one more feature and then release 0.6, without a trace of that MIT license apart from this public thread 😂 .

And, you're right: The names are more or less for insiders. My perception was that people interested in the Zettelkasten method would pretty much all be gathered around zettelkasten.de and so they should be in the know. 

I just added `wiki` to the list of tags on github `note-taking zettelkasten markdown-editor markdown`. I thought that would be enough. I am a bit hesitant with `wiki` in the name. Wikis are not a Zettelkasten... At least not your typical Web Wiki. But I am all for a name change. If one comes up with a cool name, I'll probably rename Sublimeless_ZK---anything a bit more fancy than "Markdown Zettelkasten"?
